### PR TITLE
feat: add web ui

### PR DESCRIPTION
**Pull request by learnerljh**
# Description

I developed a web UI for any-to-any model inference inspired by LLaVA.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #1314520` if this solves the issue #15213
solved #2 
- [x] I have raised an issue to propose this change ([required](https://github.com/PKU-Alignment/align-anything/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/PKU-Alignment/align-anything/blob/HEAD/.github/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -51,6 +51,10 @@ dependencies = [
     "pre-commit",
     "pytest",
     "shortuuid",
+    "gradio",
+    "uvicorn",
+    "fastapi",
+    "PIL"
```
Comment (Gaiejj) [inline pyproject.toml:57]: add `,`

File: align_anything/serve/controller.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+def heart_beat_controller(controller):
+    while True:
+        time.sleep(CONTROLLER_HEART_BEAT_EXPIRATION)
+        controller.remove_stable_workers_by_expiration()
+
+
+class Controller:
+    def __init__(self, dispatch_method: str):
+        # Dict[str -> WorkerInfo]
```
Comment (Gaiejj) [inline align_anything/serve/controller.py:76]: Can we remove this kind of annotation ?

File: align_anything/serve/controller.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+            ret = {
+                "text": server_error_msg,
+                "error_code": 2,
+            }
+            yield json.dumps(ret).encode() + b"\0"
+
+        try:
+            response = requests.post(worker_addr + "/worker_generate_stream",
+                json=params, stream=True, timeout=5)
+            # for chunk in response.iter_lines(decode_unicode=False, delimiter=b"\0"):
```
Comment (Gaiejj) [inline align_anything/serve/controller.py:226]: Please also remove this kind of annotation.

File: README.md [hunk 4]
```diff
@@ -292,23 +292,39 @@ class PKUSafeRLHF(Template):
...
-python3 -m align_anything.serve.cli --model_name_or_path your_model_name_or_path
+To launch a Gradio demo locally, follow these steps by running the commands one by one. If you intend to launch multiple model workers to compare different checkpoints, you only need to launch the controller and the web server *ONCE*.
+
+#### Launch a controller
+```Shell
+python -m align_anything.serve.controller --host 0.0.0.0 --port 10000
 ```
 
-![cli_demo](assets/cli_demo.gif)
+#### Launch a gradio web server.
```
Comment (Gaiejj) [inline README.md:306]: Please remove `.`.

Comment (learnerljh) [reply README.md:306]: Thank you!

**Review by Gaiejj — CHANGES_REQUESTED** [review]
Mainly looks good to me, but some typos exist. Please fix them.

**Review by Gaiejj — APPROVED** [review]
LGTM.

**Review by cby-pku — APPROVED** [review]

**Pull request merged**